src/o365/graph/commands/teams/teams-user-list.ts

```diff
@@ -0,0 +1,146 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+import { GroupUser } from '../o365group/GroupUser';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  role?: string;
+}
+
+
+class TeamsUserListCommand extends GraphItemsListCommand<GroupUser> {
+  public get name(): string {
+    return `${commands.TEAMS_USER_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Lists Microsoft Teams teams users for a specified team';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    this.getOwners(cmd, args.options.teamId)
+      .then((): Promise<any> => {
+        if (args.options.role != "Owner") {
+
+          return this.getMembersAndGuests(cmd, args.options.teamId)
+            .then((): Promise<any> => {
+
+              // Filter out duplicate added values for owners (as they are returned as members as well)
+              this.items = this.items.filter((groupUser, index, self) =>
```

Comment:
Nested promises complicate reading the code, we should chain them instead having a flatter code structure.